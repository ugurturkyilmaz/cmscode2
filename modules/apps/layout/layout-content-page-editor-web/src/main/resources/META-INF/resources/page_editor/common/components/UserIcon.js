/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayIcon from '@clayui/icon';
import ClaySticker from '@clayui/sticker';
import PropTypes from 'prop-types';
import React from 'react';

export default function UserIcon({fullName = '', portraitURL = '', userId}) {
	const stickerColor = parseInt(userId, 10) % 10;

	return (
		<ClaySticker
			className={`flex-shrink-0 sticker-use-icon user-icon-color-${stickerColor}`}
			displayType="secondary"
			shape="circle"
			size="lg"
		>
			{portraitURL ? (
				<div className="sticker-overlay">
					<img
						alt={`${fullName}.`}
						className="sticker-img"
						src={portraitURL}
					/>
				</div>
			) : (
				<ClayIcon symbol="user" />
			)}
		</ClaySticker>
	);
}

UserIcon.propTypes = {
	fullName: PropTypes.string,
	portraitURL: PropTypes.string,
	userId: PropTypes.string.isRequired,
};
