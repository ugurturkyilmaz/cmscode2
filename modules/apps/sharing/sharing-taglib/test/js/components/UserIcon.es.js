/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import UserIcon from '../../../src/main/resources/META-INF/resources/collaborators/js/components/UserIcon.es';

describe('UserIcon', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {asFragment} = render(
			<UserIcon fullName="Jane Smith" userId="1000" />
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
