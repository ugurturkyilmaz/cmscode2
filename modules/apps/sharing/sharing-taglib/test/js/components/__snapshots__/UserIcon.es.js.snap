// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIcon renders 1`] = `
<DocumentFragment>
  <span
    class="sticker sticker-use-icon user-icon-color-0 sticker-circle sticker-secondary sticker-lg"
  >
    <span
      class="sticker-overlay"
    >
      <svg
        class="lexicon-icon lexicon-icon-user"
        role="presentation"
      >
        <use
          xlink:href="#user"
        />
      </svg>
    </span>
  </span>
</DocumentFragment>
`;
